etcdserver/auth/auth.go

```diff
@@ -281,11 +285,14 @@ func (s *store) UpdateUser(user User) (User, error) {
 		return old, err
 	}
 
-	hash, err := s.HashPassword(user.Password)
-	if err != nil {
-		return old, err
+	if user.Password != "" {
+		var hash string
+		hash, err = s.HashPassword(user.Password)
```

Comment:
i think merge should handle this? Does the behavior of hashpassword changed somehow?